dev-tools/v2tool/server/ca.go

```diff
@@ -0,0 +1,160 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package server
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"log"
+	"math/big"
+	"time"
+)
+
+// CertificateAuthority is an abstraction for common certificate authority
+// unique for process
+type CertificateAuthority struct {
+	caCert     *x509.Certificate
+	privateKey crypto.PrivateKey
+	caPEM      []byte
+}
+
+// Pair is a x509 Key/Cert pair
+type Pair struct {
+	Crt         []byte
+	Key         []byte
+	Certificate *tls.Certificate
+}
+
+// NewCA creates a new certificate authority capable of generating child certificates
+func NewCA() (*CertificateAuthority, error) {
+	ca := &x509.Certificate{
+		DNSNames:     []string{"localhost"},
+		SerialNumber: big.NewInt(1653),
+		Subject: pkix.Name{
+			Organization: []string{"elastic-fleet"},
+			CommonName:   "localhost",
```

Comment:
So, this entire file is ripped from elastic-agent, which I couldn't import because it was in `internal/` Not quite sure what to do with it. I figure once this PR gets farther along, I'll put in a PR to elastic0-agent to move the file out of internal.

Reply:
I'm not sure if you'll be able to import `elastic-agent` without creating an import cycle since elastic-agent already [depends on](https://github.com/elastic/elastic-agent/blob/270295f4e334931df41b47e6607c238a8d6e4b44/go.mod#L15) `elastic-agent-client`

Reply:
Depending on the package it's possible.

But that might be an indication that either this code needs to be fully independent from the Elastic-Agent or be moved there.

Anyway, once we work more on the developer experience, we'll find out what is the best place for it.